2009-11-19  Aurelien Jarno  <aurelien@aurel32.net>
    
	* io/fcntl.h: Move AT_* defines to...
	* sysdeps/mach/hurd/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/i386/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/ia64/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/powerpc/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/s390/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/sh/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/sparc/bits/fcntl.h: ...here.
	* sysdeps/unix/sysv/linux/x86_64/bits/fcntl.h: ...here.

2009-11-19  Aurelien Jarno  <aurelien@aurel32.net>
    
	* sysdeps/unix/sysv/linux/alpha/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/am33/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/arm/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/cris/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/hppa/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/m68k/bits/fcntl.h: Define AT_*
	constants.
	* sysdeps/unix/sysv/linux/mips/bits/fcntl.h: Define AT_*
	constants.

---
 io/fcntl.h                                       |   14 --------------
 ports/sysdeps/unix/sysv/linux/alpha/bits/fcntl.h |   15 +++++++++++++++
 ports/sysdeps/unix/sysv/linux/am33/bits/fcntl.h  |   17 +++++++++++++++++
 ports/sysdeps/unix/sysv/linux/arm/bits/fcntl.h   |   15 +++++++++++++++
 ports/sysdeps/unix/sysv/linux/cris/bits/fcntl.h  |   15 +++++++++++++++
 ports/sysdeps/unix/sysv/linux/hppa/bits/fcntl.h  |   15 +++++++++++++++
 ports/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h  |   15 +++++++++++++++
 ports/sysdeps/unix/sysv/linux/mips/bits/fcntl.h  |   15 +++++++++++++++
 sysdeps/mach/hurd/bits/fcntl.h                   |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/i386/bits/fcntl.h        |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/ia64/bits/fcntl.h        |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/powerpc/bits/fcntl.h     |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/s390/bits/fcntl.h        |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/sh/bits/fcntl.h          |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/sparc/bits/fcntl.h       |   15 +++++++++++++++
 sysdeps/unix/sysv/linux/x86_64/bits/fcntl.h      |   15 +++++++++++++++
 16 files changed, 227 insertions(+), 14 deletions(-)

--- a/io/fcntl.h
+++ b/io/fcntl.h
@@ -102,20 +102,6 @@
 # define SEEK_END	2	/* Seek from end of file.  */
 #endif	/* XPG */
 
-#ifdef __USE_ATFILE
-# define AT_FDCWD		-100	/* Special value used to indicate
-					   the *at functions should use the
-					   current working directory. */
-# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
-# define AT_REMOVEDIR		0x200	/* Remove directory instead of
-					   unlinking file.  */
-# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
-# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
-					   traversal.  */
-# define AT_EACCESS		0x200	/* Test access permitted for
-					   effective IDs, not real IDs.  */
-#endif
-
 /* Do the file control operation described by CMD on FD.
    The remaining arguments are interpreted depending on CMD.
 
--- a/sysdeps/mach/hurd/bits/fcntl.h
+++ b/sysdeps/mach/hurd/bits/fcntl.h
@@ -220,3 +220,18 @@
 # define POSIX_FADV_DONTNEED	4 /* Don't need these pages.  */
 # define POSIX_FADV_NOREUSE	5 /* Data will be accessed once.  */
 #endif
+
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
--- a/sysdeps/unix/sysv/linux/i386/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/i386/bits/fcntl.h
@@ -234,6 +234,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/ia64/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/ia64/bits/fcntl.h
@@ -229,6 +229,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/powerpc/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/powerpc/bits/fcntl.h
@@ -234,6 +234,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/s390/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/s390/bits/fcntl.h
@@ -255,6 +255,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/sh/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/sh/bits/fcntl.h
@@ -235,6 +235,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/sparc/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/sparc/bits/fcntl.h
@@ -246,6 +246,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/sysdeps/unix/sysv/linux/x86_64/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/x86_64/bits/fcntl.h
@@ -249,6 +249,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/ports/sysdeps/unix/sysv/linux/alpha/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/alpha/bits/fcntl.h
@@ -225,6 +225,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/ports/sysdeps/unix/sysv/linux/am33/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/am33/bits/fcntl.h
@@ -179,3 +179,20 @@
 # define POSIX_FADV_DONTNEED	4 /* Don't need these pages.  */
 # define POSIX_FADV_NOREUSE	5 /* Data will be accessed once.  */
 #endif
+
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
+
--- a/ports/sysdeps/unix/sysv/linux/arm/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/arm/bits/fcntl.h
@@ -234,6 +234,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/ports/sysdeps/unix/sysv/linux/cris/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/cris/bits/fcntl.h
@@ -182,6 +182,21 @@
 # define POSIX_FADV_NOREUSE	5 /* Data will be accessed once.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 /* Provide kernel hint to read ahead.  */
--- a/ports/sysdeps/unix/sysv/linux/hppa/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/hppa/bits/fcntl.h
@@ -235,6 +235,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/ports/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h
@@ -233,6 +233,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
--- a/ports/sysdeps/unix/sysv/linux/mips/bits/fcntl.h
+++ b/ports/sysdeps/unix/sysv/linux/mips/bits/fcntl.h
@@ -247,6 +247,21 @@
 # define SPLICE_F_GIFT		8	/* Pages passed in are a gift.  */
 #endif
 
+/* Values for `*at' functions.  */
+#ifdef __USE_ATFILE
+# define AT_FDCWD		-100	/* Special value used to indicate
+					   the *at functions should use the
+					   current working directory. */
+# define AT_SYMLINK_NOFOLLOW	0x100	/* Do not follow symbolic links.  */
+# define AT_REMOVEDIR		0x200	/* Remove directory instead of
+					   unlinking file.  */
+# define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+# define AT_EACCESS		0x200	/* Test access permitted for
+					   effective IDs, not real IDs.  */
+#endif
+
 __BEGIN_DECLS
 
 #ifdef __USE_GNU
